cmd/list_subcmd.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"log"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/turbot/steampipe/pkg/cmdconfig"
+	"github.com/turbot/steampipe/pkg/constants"
+	"github.com/turbot/steampipe/pkg/display"
+	"github.com/turbot/steampipe/pkg/error_helpers"
+	"github.com/turbot/steampipe/pkg/steampipeconfig/modconfig"
+	"github.com/turbot/steampipe/pkg/types"
+	"github.com/turbot/steampipe/pkg/workspace"
+)
+
+type listSubCmdOptions struct {
+	shortDescription string
+	longDescription  string
+	allDescription   string
+	parent           *cobra.Command
+}
+
+func listSubCmd(opts listSubCmdOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:              "list",
+		TraverseChildren: true,
+		Args:             cobra.NoArgs,
+		Run:              listSubCmdRunner(opts),
+		Short:            opts.shortDescription,
+		Long:             opts.longDescription,
+	}
+
+	cmdconfig.
+		OnCmd(cmd).
+		AddBoolFlag(constants.ArgAll, "", false, opts.allDescription)
+
+	return cmd
+}
+
+func listSubCmdRunner(opts listSubCmdOptions) func(cmd *cobra.Command, args []string) {
+	if opts.parent == nil {
+		// we can panic here, since this will always come up during development
+		panic("parent is required")
+	}
+	return func(cmd *cobra.Command, args []string) {
+		parent := cmd.Parent().Name()
+		cmdToTypeMapping := map[string][]string{
+			"check": {"benchmark", "control"},
+		}
+		typ, found := cmdToTypeMapping[parent]
```

Comment:
PLEASE can you get into the habit of avoiding these kind of abbreviated names  - I've raised this so many times :(

typ -> resourceTypesToList